### PR TITLE
reworks bluespace miners

### PR DESCRIPTION
**Pull request by bloons3**
## About The Pull Request

Bluespace miners are pretty cool, this makes them give more rewards n junk

## Why It's Good For The Game

People need to understand that faustian bargains have a cost.

## Changelog
:cl:

refactor: refactored some bluespace miner code
/:cl:

### CODE REVIEW

File: code/game/objects/items/circuitboards/machine_circuitboards.dm [hunk 1]
```diff
@@ -804,12 +804,12 @@
 	icon_state = "science"
 	build_path = /obj/machinery/mineral/bluespace_miner
 	req_components = list(
-		/obj/item/stock_parts/capacitor/quadratic = 1,
+		/obj/item/stock_parts/capacitor = 1,
 		/obj/item/stock_parts/matter_bin = 3,
 		/obj/item/stock_parts/micro_laser = 1,
-		/obj/item/stock_parts/manipulator/femto = 3,
```
Comment (Sinestia) [inline code/game/objects/items/circuitboards/machine_circuitboards.dm:810]: Sneaking in the buff I see, pretty Sus behavior.

**Review by Sinestia — DISMISSED** [review]
What did it cost?

Everything.

**Comment by Sinestia** [comment]
This might warrant adding the autodoc back, as it leaves only one survivor who can't implant wings and x-ray eyes on himself, for what else is there to print.

File: code/modules/mining/machine_bluespaceminer.dm [hunk 2]
```diff
@@ -36,5 +36,16 @@
 	var/datum/component/material_container/mat_container = materials.mat_container
 	if(!mat_container || panel_open || !powered())
 		return
-	var/datum/material/ore = pick(ore_rates)
-	mat_container.insert_amount_mat((ore_rates[ore] * 1000), ore)
+
+	// profit requires a sacrifice
+	var/victims = get_living_station_crew()
+	if(length(victims)<=1) // https://youtu.be/1FzZ7llgeAU
```
Comment (Archanial) [inline code/modules/mining/machine_bluespaceminer.dm:42]:
```suggestion
	if(LAZYLEN(victims) <= 1) // https://youtu.be/1FzZ7llgeAU
```

File: code/modules/mining/machine_bluespaceminer.dm [hunk 3]
```diff
@@ -36,5 +36,16 @@
...
+		return
+
+	var/mob/living/carbon/human/victim = pick(victims)
+	victim.adjustBruteLoss(5000)
+	victim.adjustFireLoss(5000)
```
Comment (Archanial) [inline code/modules/mining/machine_bluespaceminer.dm:47]:
```suggestion
	victim.adjustBruteLoss(5000)
	victim.adjustFireLoss(5000)
	victin.Gib()
```

Comment (Archanial) [reply code/modules/mining/machine_bluespaceminer.dm:47]: actually you should make it cross server too

**Comment by ItsTisme** [comment]
jokes on you iam alone on LRP anyway free mats 4 days 😎

File: code/modules/mining/machine_bluespaceminer.dm [hunk 4]
```diff
@@ -1,6 +1,6 @@
 /obj/machinery/mineral/bluespace_miner
 	name = "bluespace mining machine"
-	desc = "A machine that uses the magic of Bluespace to slowly generate materials and add them to a linked ore silo."
+	desc = "A machine that uses the magic of Satan to slowly generate materials and add them to a linked ore silo."
 	icon = 'icons/obj/machines/mining_machines.dmi'
 	icon_state = "bs_miner"
 	density = TRUE
```
Comment (yorii) [inline code/modules/mining/machine_bluespaceminer.dm:6]: var/timer = 0

File: code/modules/mining/machine_bluespaceminer.dm [hunk 5]
```diff
@@ -36,5 +36,16 @@
 	var/datum/component/material_container/mat_container = materials.mat_container
 	if(!mat_container || panel_open || !powered())
 		return
-	var/datum/material/ore = pick(ore_rates)
-	mat_container.insert_amount_mat((ore_rates[ore] * 1000), ore)
+
```
Comment (yorii) [inline code/modules/mining/machine_bluespaceminer.dm:39]:
```suggestion
timer++
if(timer < 20) // Just guessing a value
	return
timer = 0
```

File: code/modules/mining/machine_bluespaceminer.dm [hunk 6]
```diff
@@ -1,6 +1,6 @@
 /obj/machinery/mineral/bluespace_miner
 	name = "bluespace mining machine"
-	desc = "A machine that uses the magic of Bluespace to slowly generate materials and add them to a linked ore silo."
+	desc = "A machine that uses the magic of Satan to slowly generate materials and add them to a linked ore silo."
```
Comment (yorii) [inline code/modules/mining/machine_bluespaceminer.dm:3]:
```suggestion
	desc = "A machine that uses the magic of Satan to slowly generate materials and add them to a linked ore silo."
	timer = 0
```

**Comment by jlsnow301** [comment]
component upgrades should kill crew faster imo

**Comment by eeSPee** [comment]
Might as well turn it into a superrecycler that turns sentient humans into materials...

**Comment by ItsTisme** [comment]
Would using this count as self antaging?

**Comment by Sinestia** [comment]
> Would using this count as self antaging?

The greatest sacrifices require the greatest wills

**Comment by TheFakeElon** [comment]
> refactor: refactored some bluespace miner code

Refactors are purely code changes with no gameplay effects. This is a rework

**Review by utilbird — APPROVED** [review]
Nice to see some much needed optimization

**Pull request closed without merging**